imap_processing/mag/l1a/mag_l1a_data.py

```diff
@@ -429,6 +479,478 @@ def to_signed16(n: int) -> int:
+                )
+                if seq_val >= end_primary_vector + uncompressed_vector_size + 2:
+                    # Split just after the uncompressed secondary vector
+                    secondary_boundaries = [end_primary_vector + uncompressed_vector_size + 2]
+                    # We have found the first secondary vector
+                    secondary_boundaries += [seq_val+1]
+                    vector_count += 1
+
+            # If we're greater than primary_count, we are in the secondary vectors.
+            # Like before, we skip indices with 3 ones.
+            if (
+                vector_count > primary_count
+                and seq_val - secondary_boundaries[-1] > 0
+            ):
+                secondary_boundaries.append(seq_val + 1)
+                # We have the start of the secondary vectors in
+                # secondary_boundaries, so we need to subtract one to determine
+                # the vector count. (in primary_boundaries we know we start at 0.)
+                if (len(secondary_boundaries) - 1) % 3 == 0:
+                    vector_count += 1
+                    if secondary_boundaries[-1] - secondary_boundaries[-4] > 60:
+                        # The rest of the secondary values are uncompressed.
+                        vector_count = primary_count + secondary_count + 1
+
+        # Split along the boundaries of the primary vectors. This gives us a list of
+        # bit arrays, each corresponding to a primary value (1/3 of a vector).
+        primary_split_bits = np.split(
+            vector_bits,
+            primary_boundaries,
+        )[:-1]
+        # print(f"Primary split bits: {primary_split_bits}")
+        # print(f"First vector: {first_vector}")
+
+        vector_diffs = list(map(MagL1a.decode_fib_zig_zag, primary_split_bits))
+
+        primary_vectors = MagL1a.accumulate_vectors(
+            first_vector, vector_diffs, primary_count
+        )
+
+        # If we are missing any vectors from primary_split_bits, we know we have
+        # uncompressed vectors to process.
+        primary_vector_missing = primary_count - math.ceil(len(primary_split_bits) / 3) - 1
+        vector_index = primary_count - primary_vector_missing
+        if primary_vector_missing:
+            primary_end = (
+                primary_boundaries[-1]
+                + uncompressed_vector_size * primary_vector_missing
+            )
+            uncompressed_vectors = vector_bits[primary_boundaries[-1] : primary_end]
+            for i in range(0, len(uncompressed_vectors), uncompressed_vector_size):
+                decoded_vector = MagL1a.unpack_one_vector(
+                    uncompressed_vectors[i : i + uncompressed_vector_size],
+                    compression_width,
+                    False,
+                )
+                primary_vectors[vector_index] = decoded_vector
+                primary_vectors[vector_index][3] = first_vector[3]
+                vector_index += 1
```

Comment:
Yeah, bad phrasing. They aren't missing from the data, but rather missing from the compressed vectors. If we finish with the compressed vectors, and we don't have the number that we expect, then this code assumes that they are missing because they are in uncompressed form. So, it's more like they're missing from what has been processed so far. I will try to rename this and expand my comments.